crates/zk-core/src/check.rs

```diff
@@ -96,28 +100,53 @@ impl Check {
 
         let chi = Block::try_from(&transcript.finalize().as_bytes()[..16])
             .expect("block should be 16 bytes");
-        let chis = self.compute_chis(chi);
         let macs = mem::take(&mut self.triples);
+
+        // Computation with pre-split lanes.
+        const PARALLELISM: usize = 16;
+        let n = macs.len();
+        let segment_size = n.div_ceil(PARALLELISM);
+        let starts = Self::compute_chi_starts(chi);
```

Comment:
Couple comments:

- Hardcoding a "parallelism" number doesn't seem right. The goal is so saturate the CPU cores while also optimizing for cache efficiency. So we should choose a chunk size, e.g. 1024, which is large enough to saturate caches but small enough to take advantage of work stealing.
- We don't need multiple PRGs, ChaCha12 already has a `set_stream` API for producing independent streams.

```suggestion
        const CHUNK_SIZE: usize = 1024;
        let seed = *transcript.finalize().as_bytes();
        let rng = ChaCha12Rng::from_seed(seed);
```